hashCheck/hashCheck_test.go

```diff
@@ -0,0 +1,340 @@
+				filepath.FromSlash("/home/user/documents/file2.txt"),
+			},
+			filepath.FromSlash("/home/user/images"): {
+				filepath.FromSlash("/home/user/images/image1.png"),
+				filepath.FromSlash("/home/user/images/image2.png"),
+				filepath.FromSlash("/home/user/images/image3.png"),
+			},
+		}
+
+		output := groupByDirectory(inputs)
+
+		if diff := cmp.Diff(output, expectedOutput); diff != "" {
+			t.Errorf("(-got +want)\n%s", diff)
+		}
+	})
+
+	t.Run("Should return empty map when no inputs", func(t *testing.T) {
+		var inputs []string
+
+		expectedOutput := map[string][]string{}
+
+		output := groupByDirectory(inputs)
+
+		if diff := cmp.Diff(output, expectedOutput); diff != "" {
+			t.Errorf("(-got +want)\n%s", diff)
+		}
+	})
+
+	t.Run("Should handle files in root directory", func(t *testing.T) {
+		inputs := []string{
+			filepath.FromSlash("/home/user/documents/file1.txt"),
+			filepath.FromSlash("/home/user/documents/file2.txt"),
+			filepath.FromSlash("/home/user/images/image1.png"),
+			filepath.FromSlash("/home/user/images/image2.png"),
+			filepath.FromSlash("/home/user/images/image3.png"),
+		}
+
+		expectedOutput := map[string][]string{
+			filepath.FromSlash("/home/user/documents"): {
+				filepath.FromSlash("/home/user/documents/file1.txt"),
+				filepath.FromSlash("/home/user/documents/file2.txt"),
+			},
+			filepath.FromSlash("/home/user/images"): {
+				filepath.FromSlash("/home/user/images/image1.png"),
+				filepath.FromSlash("/home/user/images/image2.png"),
+				filepath.FromSlash("/home/user/images/image3.png"),
+			},
+		}
+
+		actualOutputs := groupByDirectory(inputs)
+
+		if diff := cmp.Diff(actualOutputs, expectedOutput); diff != "" {
+			t.Errorf("(-got +want)\n%s", diff)
+		}
+	})
+}
+
+func TestParseJsonModel(t *testing.T) {
+	t.Run("Should return json model", func(t *testing.T) {
+		input := "TestParseJsonModel/valid"
+		expectedOutput := Hashes{
+			"repository/repository.go": {
+				Hash:  "94d34c809fe95f47f6157330958e673cb19922c11091af2d3259e795a47914cc",
+				GoSum: "87e39adc6f6fbe5f5b49ac3e7ce96b1193fe571760097e40f02768d43601b141",
+			},
+		}
+
+		actualOutput, err := parseJsonModel(input)
+
+		if err != nil {
+			t.Errorf("Expected nil but got %v", err)
+		}
+
+		if diff := cmp.Diff(actualOutput, expectedOutput); diff != "" {
+			t.Errorf("(-got +want)\n%s", diff)
+		}
+
+	})
+
+	t.Run("Should return error because json file is not valid", func(t *testing.T) {
+		input := "TestParseJsonModel/invalid"
+		_, err := parseJsonModel(input)
+
+		if err == nil {
+			t.Errorf("Expected error but got nil")
+		}
+	})
+
+	t.Run("Should return empty struct when file is not found", func(t *testing.T) {
+		input := "notfound"
+		expectedOutput := Hashes{}
+
+		output, err := parseJsonModel(input)
+
+		if err != nil {
+			t.Errorf("Expected nil but got %v", err)
+		}
+
+		if diff := cmp.Diff(output, expectedOutput); diff != "" {
+			t.Errorf("(-got +want)\n%s", diff)
```

Comment:
stretchify `.RequireEqual` implements this logic, feel free to use it if you want